main.tf

```diff
@@ -23,7 +23,7 @@ resource "google_storage_bucket" "bucket" {
       condition {
         age                        = try(lifecycle_rule.value.condition.age, null)
         created_before             = try(lifecycle_rule.value.condition.created_before, null)
-        with_state                 = try(lifecycle_rule.value.condition.with_state, null)
+        with_state                 = try(upper(lifecycle_rule.value.condition.with_state), null)
```

Comment:
we should not silently transform user input.... later state can be "LIVEv2" whatever and we will not be able to change this... better add validation of specific values.

Reply:
ok, will add validation in a follow up task and get rid of this now

Reply:
resolved